pandas/core/frame.py

```diff
@@ -9915,6 +9915,7 @@ def _dict_round(df: DataFrame, decimals):
                     yield _series_round(vals, decimals[col])
                 except KeyError:
                     yield vals
+            df._clear_item_cache()
```

Comment:
why is this needed?

Reply:
Oops, this snuck in. I'm going to make another PR for this separately.